### PR TITLE
remove deprecated url encoding library

### PR DESCRIPTION
**Pull request by coleshaw**
Small change, since the Magma logs are super noisy. We should remove usage of this obsolete library anyways, as we did in #1005 and #928.

### CODE REVIEW

File: magma/lib/magma/storage.rb [hunk 1]
```diff
@@ -51,14 +53,18 @@ def hmac_url(method, host, path, expiration=0, headers={})
...
             expiration: (Time.now + expiration).iso8601,
             nonce: Magma.instance.sign.uid,
             id: :magma,
             headers: headers
           ).url_params
         )
       end
+
+      def encode_path(path)
+        path.split('/').map { |c| Addressable::URI.normalized_encode(c) }.join('/')
```
Comment (graft) [inline magma/lib/magma/storage.rb:66]: Somewhere we use URI.encode_www_form_component here, what is the difference?

**Review by graft — APPROVED** [review]

**Comment by coleshaw** [comment]
> Err, just recalled this issue, which may cause trouble for files with a ? in them:
> 
> ```
> pry(#<FolderController>)> Addressable::URI.normalized_encode("blah?blah")
> => "blah?blah"
> ```

Hm...I believe you are right, that may cause problems, but is that not the same behavior as `URI.encode`?

```
irb(main):001:0> URI.encode("blah?blah")
(irb):1: warning: URI.escape is obsolete
=> "blah?blah"
```

**Comment by coleshaw** [comment]
Perhaps that scenario also has implied consequences for other places in the code, too, not just here...

**Comment by graft** [comment]
However:
```
irb(main):003:0> URI.encode_www_form_component("blah?blah")
=> "blah%3Fblah"
```

**Comment by coleshaw** [comment]
Yes, that's right. So the problem is ... for some characters, `encode_www_form_component` does what we want (`?`), because it doesn't assume the `?` is the query character like the other methods, and thus encodes it. But on the other hand, ` ` is encoded incorrectly as if for a form and not a URL path, which breaks for us because the file name becomes part of the URL path ... :thinking:

**Comment by coleshaw** [comment]
I think trying to account for that difference will require changes to other parts of the code, too, not just removing this one deprecated method call.

**Comment by coleshaw** [comment]
I wonder if we just need our own encoding methods to account for the ` ` as a separate use case on top of `encode_www_form_component`. As far as I know, that might be the one character that behaves differently, depending on if it's in the query params or URL path?

**Comment by coleshaw** [comment]
Okay, tweaked this a bit to have the encoding parts use a custom encoder, based on `URI.encode_www_form_component`, but then making sure any stray `+` are converted into the URL-friendly `%20`. Hopefully that is the only special character that we'll have to deal with ...

**Pull request merged**